CMakeLists.txt

```diff
@@ -2,284 +2,35 @@
-# setup user specified installation directory if any, regardless previous platform default settings
-if (CMAKE_INSTALL_BINDIR)
-    set(BINARY_DIRECTORY "${CMAKE_INSTALL_BINDIR}")
-endif()
-
-if (CMAKE_INSTALL_LIBDIR)
-    set(LIBRARY_DIRECTORY "${CMAKE_INSTALL_LIBDIR}")
-endif()
-
-if (CMAKE_INSTALL_INCLUDEDIR)
-    set(INCLUDE_DIRECTORY "${CMAKE_INSTALL_INCLUDEDIR}")
-endif()
-
-if(BUILD_SHARED_LIBS)
-    set(ARCHIVE_DIRECTORY "${BINARY_DIRECTORY}")
-else()
-    set(ARCHIVE_DIRECTORY "${LIBRARY_DIRECTORY}")
-endif()
-
-if (ENABLE_ADDRESS_SANITIZER)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address -g -fno-omit-frame-pointer")
-    if (CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 7.1)
-        message(STATUS "adding libasan as static explicitly for GCC 7+")
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -static-libasan")
-    endif()
-endif()
-
-include(CheckCXXSymbolExists)
-
-check_cxx_symbol_exists("pathconf" "unistd.h" HAS_PATHCONF)
-if (HAS_PATHCONF)
-    add_definitions(-DHAS_PATHCONF)
-endif()
-
-check_cxx_symbol_exists("umask" "sys/stat.h" HAS_UMASK)
-if (HAS_UMASK)
-    add_definitions(-DHAS_UMASK)
-endif()
-
-add_sdks()
-
-# for user friendly cmake usage
-include(setup_cmake_find_module)
-
-# for generating make uninstall target
-if (NOT TARGET uninstall)
-    ADD_CUSTOM_TARGET(uninstall "${CMAKE_COMMAND}" -P "${AWS_NATIVE_SDK_ROOT}/cmake/make_uninstall.cmake")
-else()
-    ADD_CUSTOM_TARGET(uninstall-awssdk "${CMAKE_COMMAND}" -P "${AWS_NATIVE_SDK_ROOT}/cmake/make_uninstall.cmake")
-endif()
+cmake_minimum_required(VERSION 3.12 FATAL_ERROR)
+
+option(LEGACY_BUILD "If enabled, the SDK will use 1.9 version of CMake files to build" ON)
+if (LEGACY_BUILD)
+    message(WARNING "In 1.10.0 release, our CMake build scripts are been updated to provide modern, more stable, CMake "
+            "builds. The legacy support fallbacks to the 1.9 CMake scripts to allow customers to upgrade their "
+            "building scripts at their own pace. "
+            "The legacy support is set by default in 1.10.0, when you complete build updating scripts please "
+            "update the build flags as mentioned in README.md and set -DLEGACY_BUILD=OFF. "
+            "The legacy support will be removed at 1.11.0 release.")
+    list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake_legacy")
+    include(initialize_project_version)
+    project("aws-cpp-sdk-all" VERSION "${PROJECT_VERSION}" LANGUAGES CXX)
+    include(legacy_main)
+else ()
+    message(STATUS "Building with 1.10 CMake scripts.")
+    string(CONCAT DESCRIPTION_STRING "The AWS SDK for C++ provides a modern C++ (standard version C++11 or later) "
+            "interface for Amazon Web Services (AWS).")
+    project("aws-cpp-sdk"
+            LANGUAGES CXX
+            VERSION 1.10.0
+            DESCRIPTION ${DESCRIPTION_STRING}
+            HOMEPAGE_URL "https://docs.aws.amazon.com/sdk-for-cpp"
+            )
+    set(CMAKE_CXX_STANDARD 11)
```

Comment:
since we're requiring a higher cmake version should we do something crazy and update to 14 or even higher?

Reply:
Not in the plans for now.